juju/juju
# (chore): remove origin header

This PR removes the "Origin" header set in the `connectStream` used by the Juju CLI when connecting to streaming endpoints (notably /log for debug-logs).

The Origin header is primarily used by web browsers to indicate the origin (scheme, host, and port) of the request. It's a part of the browser's Cross-Origin Resource Sharing (CORS) mechanism to help servers determine the source of a request.

Servers often use this to enforce security policies, like preventing cross-origin attacks.

The CLI shouldn't be setting this header and doesn't seem to use it. 

JIMM checks this header value to protect against CORS attacks. To briefly explain why, JIMM supports different auth mechanisms and supports auth via cookies, and so needs to guard against [Cross-site WebSocket hijacking](https://portswigger.net/web-security/websockets/cross-site-websocket-hijacking).

## Checklist



~- [ ] Code style: imports ordered, good names, simple structure, etc~
~- [ ] Comments saying why design decisions were made~
~- [ ] Go unit tests, with comments saying what you're testing~
~- [ ] [Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~
~- [ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

- `make install`
- `juju bootstrap localhost test`
- `juju add-model foo`
- `juju debug-log`
- See logs are available

## Links

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin
